openc3/spec/interfaces/protocols/cobs_protocol_spec.rb

```diff
@@ -0,0 +1,178 @@
+# encoding: ascii-8bit
+
+# Copyright 2023 OpenC3, Inc.
+# All Rights Reserved.
+#
+# This program is free software; you can modify and/or redistribute it
+# under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation; version 3 with
+# attribution addendums as found in the LICENSE.txt
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# This file may also be used under the terms of a commercial license
+# if purchased from OpenC3, Inc.
+
+require 'spec_helper'
+require 'openc3/interfaces/protocols/cobs_protocol'
+require 'openc3/interfaces/interface'
+require 'openc3/streams/stream'
+
+module OpenC3
+  describe CobsProtocol do
+
+    class CobsStream < Stream
+      def connect; end
+      def connected?; true; end
+      def disconnect; end
+      def read; $buffer; end
+      def write(data); $buffer = data; end
+    end
+
+    def build_example_data
+      @examples = [
+        ["\x00", "\x01\x01\x00"],
+        ["\x00\x00", "\x01\x01\x01\x00"],
+        ["\x00\x11\x00", "\x01\x02\x11\x01\x00"],
+        ["\x11\x22\x00\x33", "\x03\x11\x22\x02\x33\x00"],
+        ["\x11\x22\x33\x44", "\x05\x11\x22\x33\x44\x00"],
+        ["\x11\x00\x00\x00", "\x02\x11\x01\x01\x01\x00"],
+      ]
+      data = ''
+      (0..254).each do |char|
+        data << [char].pack('C')
+      end
+      result = "\x01\xFF" + data[1..-1] + "\x00"
```

Comment:
These examples are all from here: https://en.wikipedia.org/wiki/Consistent_Overhead_Byte_Stuffing. I assumed this example would be 7: 
`01 02 03 ... FD FE => FF 01 02 03 ... FD FE 00`
But it looks like you're starting with 0. Could you move this to the end and put in the actual example 7 which starts at 1. Maybe even call out the webpage again at the top of your examples so we know that's where they came from.